sdk/azidentity/interactive_browser_credential.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azidentity
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/pkg/browser"
+)
+
+// InteractiveBrowserCredentialOptions can be used when providing additional credential information, such as a client secret.
+// Also use these options to modify the default pipeline behavior through the TokenCredentialOptions.
+type InteractiveBrowserCredentialOptions struct {
+	// The Azure Active Directory tenant (directory) ID of the service principal.
+	TenantID *string
+	// The client (application) ID of the service principal.
+	ClientID *string
+	// The client secret that was generated for the App Registration used to authenticate the client. Only applies for web apps.
+	ClientSecret *string
+	// The redirect URI used to request the authorization code. Must be the same URI that is configured for the App Registration.
+	RedirectURI *string
+	// Options allows configuring the management of the requests sent to Azure Active Directory.
+	Options *TokenCredentialOptions
+}
+
+// InteractiveBrowserCredential enables authentication to Azure Active Directory using an interactive browser to log in.
+type InteractiveBrowserCredential struct {
+	client *aadIdentityClient
+	// Gets the Azure Active Directory tenant (directory) ID of the service principal.
+	tenantID string
+	// Gets the client (application) ID of the service principal.
+	clientID string
+	// Gets the client secret that was generated for the App Registration used to authenticate the client.
+	clientSecret string
+	// The redirect URI used to request the authorization code. Must be the same URI that is configured for the App Registration.
+	redirectURI string
+}
+
+// DefaultInteractiveBrowserCredentialOptions use this function to get the default values for logging in with
+// an interactive browser window.
+func DefaultInteractiveBrowserCredentialOptions() *InteractiveBrowserCredentialOptions {
+	tenantID, clientID, clientSecret, redirectURI := organizationsTenantID, developerSignOnClientID, "", ""
+	return &InteractiveBrowserCredentialOptions{
+		TenantID:     &tenantID,
```

Comment:
Is it ever legal for `TenantID` or `ClientID` to be `nil`/empty string?  If they always have to have _some_ value (which the above initialization alludes to) then they probably shouldn't be pointer-to-string.